### PR TITLE
Apply PHP 8.0 Syntax and constructor promotion

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.0, php 8.0 syntax can constuctor promotion can be applied.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

File: src/Psr/CacheItemPool/CacheItem.php [hunk 1]
```diff
@@ -47,9 +42,15 @@ final class CacheItem implements CacheItemInterface
...
      */
-    public function __construct(string $key, $value, bool $isHit, ?ClockInterface $clock = null)
-    {
-        $this->key   = $key;
+    public function __construct(
+        /**
+         * Cache key
+         */
+        private string $key,
+        $value,
```
Comment (Ocramius) [inline src/Psr/CacheItemPool/CacheItem.php]: `mixed`?

File: src/Psr/CacheItemPool/CacheItem.php [hunk 2]
```diff
@@ -47,9 +42,15 @@ final class CacheItem implements CacheItemInterface
     /**
      * @param mixed $value
      */
-    public function __construct(string $key, $value, bool $isHit, ?ClockInterface $clock = null)
-    {
-        $this->key   = $key;
+    public function __construct(
+        /**
+         * Cache key
+         */
```
Comment (Ocramius) [inline src/Psr/CacheItemPool/CacheItem.php]: Can be dropped

File: src/Psr/CacheItemPool/CacheItem.php [hunk 3]
```diff
@@ -47,9 +42,15 @@ final class CacheItem implements CacheItemInterface
...
+        private string $key,
+        $value,
+        bool $isHit,
+        ?ClockInterface $clock = null
+    ) {
         $this->value = $isHit ? $value : null;
         $this->isHit = $isHit;
```
Comment (Ocramius) [inline src/Psr/CacheItemPool/CacheItem.php:47]: Can perhaps be `private bool $isHit` in the ctor?

File: src/Storage/Adapter/KeyListIterator.php [hunk 4]
```diff
@@ -52,13 +38,19 @@ class KeyListIterator implements IteratorInterface, Countable
...
-        $this->keys    = $keys;
-        $this->count   = count($keys);
+    public function __construct(
+        /**
+         * The storage instance
+         */
+        protected StorageInterface $storage,
+        /**
+         * Keys to iterate over
+         */
```
Comment (Ocramius) [inline src/Storage/Adapter/KeyListIterator.php]: These docblocks probably need to be merged with the constructor docblock

File: src/Storage/Capabilities.php [hunk 5]
```diff
@@ -155,19 +134,22 @@ class Capabilities
...
      * Constructor
-     *
-     * @param array             $capabilities
      */
     public function __construct(
-        StorageInterface $storage,
-        stdClass $marker,
+        /**
+         * The storage instance
+         */
```
Comment (Ocramius) [inline src/Storage/Capabilities.php]: Can be dropped

File: src/Storage/Capabilities.php [hunk 6]
```diff
@@ -155,19 +134,22 @@ class Capabilities
...
+        protected StorageInterface $storage,
+        /**
+         * A marker to set/change capabilities
+         */
+        protected stdClass $marker,
         array $capabilities = [],
-        ?Capabilities $baseCapabilities = null
+        /**
+         * Base capabilities
+         */
```
Comment (Ocramius) [inline src/Storage/Capabilities.php]: Can be dropped

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Psr/CacheItemPool/CacheItem.php (hunk 1 above)
Comment (samsonasik) [reply src/Psr/CacheItemPool/CacheItem.php]: updated https://github.com/laminas/laminas-cache/pull/239/commits/414c55b4ef89e16af4c9bb2b8404443e480697e3

File: src/Storage/Capabilities.php (hunk 6 above)
Comment (samsonasik) [reply src/Storage/Capabilities.php]: dropped at https://github.com/laminas/laminas-cache/pull/239/commits/0ae0e872b3052909299293c0752ca8f41a282f7f

File: src/Storage/Capabilities.php (hunk 5 above)
Comment (samsonasik) [reply src/Storage/Capabilities.php]: dropped at https://github.com/laminas/laminas-cache/pull/239/commits/0ae0e872b3052909299293c0752ca8f41a282f7f

File: src/Psr/CacheItemPool/CacheItem.php (hunk 3 above)
Comment (samsonasik) [reply src/Psr/CacheItemPool/CacheItem.php:47]: updated at https://github.com/laminas/laminas-cache/pull/239/commits/0ae0e872b3052909299293c0752ca8f41a282f7f

File: src/Storage/Adapter/KeyListIterator.php (hunk 4 above)
Comment (samsonasik) [reply src/Storage/Adapter/KeyListIterator.php]: moved to constructor docblock at https://github.com/laminas/laminas-cache/pull/239/commits/2c5581d884b3833ff3705c39f88a4a6060aa6ed0

File: src/Psr/CacheItemPool/CacheItem.php (hunk 2 above)
Comment (samsonasik) [reply src/Psr/CacheItemPool/CacheItem.php]: dropped at https://github.com/laminas/laminas-cache/pull/239/commits/cd455513e109d02a67c036a58791c657d7e6e702

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Pull request merged**